### DISCUSSION TITLE
Window update issue

### DISCUSSION THREAD

**Discussion by heinowalther** [FreeRDP/FreeRDP#6994] [General]
Hi there
I'm using FreeRDP via the Royal TSX plugin on my Mac Book.
The Specific version of FreeRDP is now specified, but the Royal TSX version is 4.3.6 (latest one)
I have the issue that when connected to a Windows host, some dialog boxes are not shows right away.. some times it requires the mouse to be moved a bit...
An example could be Total Commander, which I use mostly from keyboard... when selecting some files for movement, then pressing F6, the resulting dialog box to confirm is not shown... I can then move the mouse a bit, and it is shown... or I can just hit Enter to confirm, and the window is updated accordingly... 
This is of cause not isolated to Total Commander, but is also an issue in other applications.. if you use the mouse all the time it is not that big a problem...
I have of cause tried to adjust the connection setting, lowering the colours and resolution etc.. but it does not seem to make any difference... 
My MacBook is connected to the server via 1G ethernet... and WiFI, but also that does not make any difference.

If anyone have any suggestions, they are very welcome :-)

/Heino

**Comment by akallabeth** [comment]
This behavior depends a lot on connection settings.
Some combinations are not properly handled/are buggy (server side only what `mstsc` uses is properly tested ;))
Try the equivalent of `/network:auto`

**Reply by heinowalther** [reply]
Well in Royal TSX I cannot set this option.. there are some performance options which include:
- Desktop Background
- Desktop composition
- Menu and window animations
- Show window while dragging
- Font Smoothing
- Visual styles
- RemoteFX
- RDP8 Graphics Pipeline (with sub options)

Under connection you can specify:
- Bitmap Caching
- Glyph Caching
- Compression
- Keep-Alive

I have not tried every combination of these, but I have basically tried to turn everything on, and turn everything off...
It seems that if everything is turned off, the "pauses" where the dialog is not shown, are slightly less than if other options are enabled... at which point the delay from you realise that the dialog is not shown until is shows is about 5 secs.
So the result is very annoying, because you think you missed the keypress because nothing is shown... :-)
It's not just in Total Commander, I have also seen this in other applications.

When using the Microsoft Remote Desktop application for Mac OS, I do not have these issues... here I just connected with the defaults, and it works...  yet I would very much like to keep using the Royal TSX application if at all possible, and the only plugin there is FreeRDP... 

The Windows server I connect to is a Windows 2016 (VM running in VMWare)

Very strange issue...

**Reply by akallabeth** [reply]
I´d create an issue with `Royal TSX` then, maybe they can help.

**Reply by heinowalther** [reply]
I have, but I thought it might be worth going to the source ;-)
I have found out that the FreeRDP version used is based on this git revision: b97c6d8
How old is that revision?

**Reply by akallabeth** [reply]
That version is pre `2.0.0` so not based on one of the stable releases.

**Reply by heinowalther** [reply]
OK I was able to upgrade the Royal TSX to verison 5.0 (Beta) which ises this release 06219e4
I seems better, but I will need a bit of time, because the problem is a bit intermittent...
